index.js

```diff
@@ -180,72 +180,91 @@
   //  Value :: Location
   var Value = 'Value';
 
-  //  _funcPath :: (Boolean, Array String, a) -> Nullable Function
-  function _funcPath(allowInheritedProps, path, _x) {
-    var x = _x;
-    for (var idx = 0; idx < path.length; idx += 1) {
-      var k = path[idx];
-      if (x == null || !(allowInheritedProps || has(k, x))) return null;
-      x = x[k];
-    }
-    return typeof x === 'function' ? x : null;
+  //  prefix :: String
+  var prefix = 'fantasy-land/';
+
+  //  funcProp :: (String, a) -> Nullable Function
+  function funcProp(prop, x) {
+    return x != null && typeof x[prop] === 'function' ? x[prop] : null;
   }
 
-  //  funcPath :: (Array String, a) -> Nullable Function
-  function funcPath(path, x) {
-    return _funcPath(true, path, x);
+  //  functionImpl :: (String, String) -> Nullable Function
+  function functionImpl(type, name) {
+    return implementations[type][name];
```

Comment:
This highlights the odd nature of having `getFunction` or `getBoundMethod` return a Nullable. We assign the return `null` to a property which expects to be a function, and then let the code fail later down the line.

I think it would be better to remove the Nullable, and return a default function which throws a descriptive error message.